src/TicTacToe/EndingConditions.java

```diff
@@ -0,0 +1,57 @@
+package TicTacToe;
+
+public class EndingConditions {
+    private Board board;
+
+    public EndingConditions (Board board){
+        this.board = board;
+    }
+
+    public boolean isDraw(){
+        char elements[][] = board.getElements();
+        for (int i = 0; i < elements.length; i++) {
+            for (int j = 0; j < elements[i].length; j++) {
+                if (elements[i][j] >= '1'&& elements[i][j] <='9' || isWin()) {
```

Comment:
Chock isWin condition before all loops to skip going inside loops in that case.

Reply:
done